### PR TITLE
Port gsheet powered pages over from CF

### PR DESCRIPTION
**Pull request by uxengine**
Everything is mostly copied as is.
I've just tweaked gsheets.js so it can be plugged in via middlewares, because it needs to call server.use(...).
  
Works nicely with https://github.com/orbiting/republik-frontend/pull/41

### CODE REVIEW

File: graphql/resolvers/_mutations/submitQuestion.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  await Promise.all([
+    sendMail({
+      to: process.env.QUESTIONS_MAIL_FROM_ADDRESS,
+      fromEmail: process.env.QUESTIONS_MAIL_FROM_ADDRESS,
+      subject: 'Neue FAQ',
+      text: `${user.email}${name} hat folgende Frage gestellt:\n\n${question}`
+    }),
+    sendMailTemplate({
+      to: user.email,
+      fromEmail: process.env.QUESTIONS_MAIL_FROM_ADDRESS,
```
Comment (patte) [inline graphql/resolvers/_mutations/submitQuestion.js]: either || to DEFAULT_MAIL_FROM_ADDRESS or do checkEnv for QUESTIONS_MAIL_FROM_ADDRESS. I'd prefer the first.

File: graphql/resolvers/_mutations/submitQuestion.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      subject: t('api/faq/mail/subject'),
```
Comment (patte) [inline graphql/resolvers/_mutations/submitQuestion.js]: add translation

File: graphql/resolvers/_mutations/submitQuestion.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      templateName: 'cf_faq',
+      globalMergeVars: [
+        { name: 'NAME',
+          content: user.firstName + ' ' + user.lastName
```
Comment (patte) [inline graphql/resolvers/_mutations/submitQuestion.js]: use user.name

File: graphql/resolvers/_queries/events.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+module.exports = async (_, args, {pgdb}) => {
+  const data = await pgdb.public.gsheets.findOneFieldOnly({name: 'events'}, 'data')
+  if (!data) {
+    return []
+  }
+  return data
+    .filter(d => d.published)
+    .sort((a, b) => new Date(a.date) - new Date(b.date))
```
Comment (patte) [inline graphql/resolvers/_queries/events.js]: use d3's descending

File: graphql/resolvers/_queries/updates.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+module.exports = async (_, args, {pgdb}) => {
+  const data = await pgdb.public.gsheets.findOneFieldOnly({name: 'updates'}, 'data')
+  if (!data) {
+    return []
+  }
+  const now = new Date()
+  return data
+    .filter(d => (new Date(d.publishedDateTime) < now))
+    .sort((a, b) => new Date(b.publishedDateTime) - new Date(a.publishedDateTime))
```
Comment (patte) [inline graphql/resolvers/_queries/updates.js]: use d3's descending

File: lib/formats.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+const {timeFormatLocale} = require('d3-time-format')
```
Comment (patte) [inline lib/formats.js]: remove this file and use @orbiting/backend-modules-formats
eg. `const { timeFormat } = require('@orbiting/backend-modules-formats')`

File: lib/gsheets.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+const slugify = require('./slugify')
+
+const dateParse = utcTimeParse('%x') // %x - the locale’s date
+const dateTimeParse = timeParse('%x %H:%M')
+
+const mapping = {
+  '12oN7o21txZKxTH_RypfWLLeXFfZE2P-Yu5kg-VoHmfw': 'faqs',
+  '1Y0vFkykAHuaGzkI5PEirnoGVmAREYr9RSUcFY7_okHk': 'updates',
+  '1rktqc3xhluZLH6OrdmY456LaiCOk0L2tMD5ZjwG89lo': 'events'
+}
```
Comment (patte) [inline lib/gsheets.js]: This id's here are just ugly, shame on me.
create a env var
`GSHEETS={id1: faqs,id2:updates}`
then use it like this:
```
const { GSHEETS } = process.env
const mapping = JSON.parse(GSHEETS) <- it's desired that this explodes on app start if the json is invalid.
```

File: package.json [hunk 8]
```diff
@@ -21,6 +21,7 @@
     "d3-array": "^1.2.1",
     "d3-dsv": "^1.0.8",
     "d3-time": "^1.0.8",
+    "d3-time-format": "^2.1.1",
```
Comment (patte) [inline package.json]: remove this (@orbiting/backend-modules-formats)

File: server.js [hunk 9]
```diff
@@ -12,7 +12,10 @@ module.exports.run = () => {
   const executableSchema = makeExecutableSchema(merge(localModule, [documents]))
 
   // middlewares
-  const middlewares = [ require('./modules/crowdfundings/express/paymentWebhooks') ]
+  const middlewares = [
+    require('./modules/crowdfundings/express/paymentWebhooks'),
+    require('./lib/gsheets')
```
Comment (patte) [inline server.js]: move the file to `./express/ghseets`

**Review by patte — CHANGES_REQUESTED** [review]

**Comment by uxengine** [comment]
@patte  All comments addressed.

**Pull request merged**